packages/lesswrong/components/posts/HideRepeatedPostsContext.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { createContext, useContext, useRef } from "react";
+
+export type HideRepeatedPostsPayload = {
+  isPostRepeated: (ownerId: string, postId: string) => boolean,
+  addPost: (ownerId: string, postId: string) => void,
+}
+
+const HideRepeatedPostsContext = createContext<HideRepeatedPostsPayload>({
+  isPostRepeated: (ownerId: string, postId: string) => false,
+  addPost: (ownerId: string, postId: string) => {},
+});
+
+export const useHideRepeatedPosts = () => {
+  const id = useRef("hrp" + Math.floor(Math.random() * 1e8));
+  const {isPostRepeated, addPost} = useContext(HideRepeatedPostsContext);
+  return {
+    isPostRepeated: isPostRepeated.bind(null, id.current),
+    addPost: addPost.bind(null, id.current),
+  };
+}
+
+export const HideRepeatedPostsProvider = ({children}) => {
```

Comment:
I would add internal documentation. I figured out how this worked, but it took a bit of reading.

Reply:
Good call - done!